File: 3349-maximum-length-substring-with-two-occurrences/maximum-length-substring-with-two-occurrences.cpp
class Solution {
public:
    int maximumLengthSubstring(string s) {
        int left=0,right=0,max_len=0;
        int n =s.length();
        unordered_map <char,int>mp;
        while(right<n){
            mp[s[right]]++;
            while(mp[s[right]]>2){
                mp[s[left]]--;
                left++;
            }
            max_len=max(max_len,(right-left+1));
            right++;
        }
        return max_len;
    }
};

File: 3349-maximum-length-substring-with-two-occurrences/maximum-length-substring-with-two-occurrences.py
class Solution:
    def maximumLengthSubstring(self, s: str) -> int:
        left = right = ma_len=0
        size=len(s)
        freq_map={}
        while right<size:
            if s[right] not in freq_map:
                freq_map[s[right]]=0
            freq_map[s[right]] +=1
            while freq_map[s[right]]>2:
                freq_map[s[left]] -=1
                left+=1
            ma_len=max(ma_len,(right-left+1))
            right+=1
        
        return ma_len